src/deploy/osp_deployer/powerflexgw.py

```diff
@@ -0,0 +1,257 @@
+
+        cmd = "mkdir -p " + self.rpm_dir
+        self.run_as_root(cmd)
+        
+        logger.debug("Uploading powerflex gateway rpm")
+        source_file = "/root/JetPack/src/pilot/powerflex/rpms/" + \
+                      self.settings.powerflex_gateway_rpm
+        self.upload_file(source_file,
+                         self.rpm_dir + "/" + \
+                         self.settings.powerflex_gateway_rpm)
+
+
+    def install_gateway(self):
+  
+        logger.debug("Installing the gateway")
+        powerflexgw_ip = self.settings.powerflexgw_vm.public_api_ip
+        cmd = "GATEWAY_ADMIN_PASSWORD=" + \
+              self.settings.powerflex_password + \
+              " rpm -ivh " + \
+              self.rpm_dir + \
+              "/" + \
+              self.settings.powerflex_gateway_rpm
+        self.run_as_root(cmd)
+
+
+    def configure_gateway(self):
+
+        logger.debug("Retrieving MDM IPs")
+
+        mdm_ips = []
+        properties = '/opt/emc/scaleio/gateway/webapps/ROOT/WEB-INF/classes/gatewayUser.properties'
+        director = Director()
+
+        stor_ = self.settings.storage_network.rsplit(".", 1)[0]
+        stor_.replace(".", '\.')
+
+        re = director.run_as_root("cat /var/lib/mistral/" +
+                                  self.settings.overcloud_name + 
+                                  "/powerflex-ansible/inventory.yml " +
+                                  "| sed -n '/mdms/,/tbs/{//!p}'")
+        mdm_nodes = re[0].split("\n")
+        mdm_nodes.pop()
+
+        for each in mdm_nodes:
+            provisioning_ip = each.split(" ")[0]
+            ssh_opts = (
+                " -o StrictHostKeyChecking=no "
+                " -o UserKnownHostsFile=/dev/null "
+                " -o KbdInteractiveDevices=no")
+
+            cmd = ("ssh " + ssh_opts + " heat-admin@" +
+                   provisioning_ip +
+                   " /sbin/ifconfig | grep inet.*" +
+                   stor_ +
+                   " | awk '{print $2}'")
+            re = director.run_tty(cmd)
+
+            mdm_ip = re[0].split("\n")[1]
+            mdm_ips.append(mdm_ip.strip())
+
+        logger.debug("Updating gatewayUser.properties")
+
+        cmd = ('sed -i "s|^mdm.ip.addresses=*|' +
+               'mdm.ip.addresses=' +
+               mdm_ips[0] +
+               ',' +
+               mdm_ips[1] +
+               '|" ' +
+               properties)
+
+        self.run_as_root(cmd)
+
+
+    def get_ssl_certificates(self):
+
+        cmd = "mkdir -p " + self.certs_dir
+        self.run_as_root(cmd)
+
+        logger.debug("Retrieving all nodes IPs")
+
+        node_ips = []
+        director = Director()
+
+        cmd = "grep " + self.ip + " ~./ssh/known_hosts"
+        re = director.run(cmd)
+        
+        if self.ip not in re:
+            logger.debug("First time connection to the powerflex gateway vm")
+            cmd = "ssh-keyscan -t ecdsa " + self.ip + " >> ~/.ssh/known_hosts"
+            re = director.run(cmd)
```

Comment:
Is there a case where this would re run after the VM was already created ?

Reply:
True. Removing it.